File: AndroidBaseTemplate/app/src/main/java/io/shtanko/basetemplate/ui/activity/MainActivity.java
package io.shtanko.basetemplate.ui.activity;

import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import io.shtanko.basetemplate.R;

public class MainActivity extends BaseActivity {

  @Override protected int layoutRes() {
    return R.layout.activity_main;
  }

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

  }
}

File: AndroidStorageTemplate/app/src/main/kotlin/io/shtanko/storagetemplate/MainActivity.kt
package io.shtanko.storagetemplate

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.github.kittinunf.fuel.httpGet
import com.github.kittinunf.result.Result

class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    "https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty".httpGet().responseString { request, response, result ->

      when (result) {
        is Result.Failure -> {
          print("FAILURE")
        }
        is Result.Success -> {
          print("SUCCESS")
        }

      }
    }

  }
}